File: cpp/program47.cpp
// program165.c

#include<iostream>
//#include<string.h>
using namespace std;

class String 
{
    public:

        int CountVowels(char *str)
        {
            int iCount = 0;

            while(*str != '\0')
            {
                if ((*str == 'a') || (*str == 'e') || (*str == 'i') || (*str == 'o') || (*str == 'u') || (*str == 'A') || (*str == 'E') || (*str == 'I') || (*str == 'O') || (*str == 'U'))
                {
                    iCount++;
                }
                str++;
            }
            return iCount;
        }
};

int main()
{
    char Arr[50];
    int iRet = 0;

    cout<<"Enter string : \n";
    cin.getline(Arr,50);

    String obj;

    iRet = obj.CountVowels(Arr);
    cout<<"Count of vowels is : "<<iRet<<"\n";

    return 0;
}

File: java/program47.java
import java.util.*;

class StringX
{
    public String str;

    StringX(String srt)
    {
        this.str = srt;
    }

    public int CountVowels()
    {
        byte[] Arr = str.getBytes();
        int count = 0;

        for(int i = 0; i < Arr.length; i++)
        {
            if((Arr[i] == 'A') || (Arr[i] == 'E') || (Arr[i] == 'E') || (Arr[i] == 'I') || (Arr[i] == 'O') ||(Arr[i] == 'U') || (Arr[i] == 'a') || (Arr[i] == 'e') || (Arr[i] == 'i') || (Arr[i] == 'o') || (Arr[i] == 'u'))
            {
                count++;
            }
        }
        return count;
    }
}

class program47
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter String");
        String str = sobj.nextLine();

        StringX obj = new StringX(str);
        int iRet = obj.CountVowels();

        System.out.println("Count of the vowels is : "+iRet);
    }
}